src/Microsoft.OpenApi/Validations/OpenApiValidator.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Exceptions;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Services;
+using Microsoft.OpenApi.Validations;
+
+namespace Microsoft.OpenApi.Validations
+{
+    /// <summary>
+    /// Class containing dispatchers to execute validation rules on for Open API document.
+    /// </summary>
+    public class OpenApiValidator : OpenApiVisitorBase
```

Comment:
Normally, the OpenApiValidator should be internal.

Reply:
I think it needs to be public in this implementation though. The user needs to instantiate OpenApiWalker with new OpenApiValidator() as an argument to perform validation.

---
In reply to: [158116144](https://github.com/Microsoft/OpenAPI.NET/pull/163#discussion_r158116144) [](ancestors = 158116144)

Reply:
@PerthCharern Sorry, Yes. In Darrel PR, it's public. But I mean it should be internal if we have a OpenApiValidator.

Reply:
I'm not sure how we want to incorporate the Validator.  We could make it an internal implementation detail, or we could allow users to initiate it directly via the Validator visitor.

Reply:
If we can decide what are the use cases for validating fragments of the OpenApiDocument then we can decide what portions of the traversal algorithm we should expose as public.  It is worth noting that is much easier to add more granularity in a future release than to take it away.